pkg/apis/kni/v1alpha1/knicluster_status.go

```diff
@@ -0,0 +1,141 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+// ObjectReference contains enough information to let you inspect or modify the referred object.
+type ObjectReference struct {
```

Comment:
How about using this instead? https://github.com/kubernetes/kubernetes/blob/v1.14.3/staging/src/k8s.io/api/core/v1/types.go#L4538